src/Provider/ImageProvider.php

```diff
@@ -207,6 +208,49 @@ public function generatePrivateUrl(MediaInterface $media, $format)
         return $this->thumbnail->generatePrivateUrl($this, $media, $format);
     }
 
+    public function validate(ErrorElement $errorElement, MediaInterface $media)
+    {
+        parent::validate($errorElement, $media);
+
+        if (null === $media->getBinaryContent()) {
+            return;
+        }
+
+        $minWidth = 0;
+        $minHeight = 0;
+
+        foreach ($this->getFormatsForContext($media->getContext()) as $format) {
+            if (!$format['constraint']) {
+                continue;
+            }
+
+            $minWidth = max($minWidth, $format['width'] ?? 0);
+            $minHeight = max($minHeight, $format['height'] ?? 0);
+        }
+
+        if (0 === $minWidth && 0 === $minHeight) {
+            return;
+        }
+
+        try {
+            $image = $this->imagineAdapter->open($media->getBinaryContent()->getPathname());
+        } catch (\RuntimeException $e) {
+            return;
+        }
+
+        $size = $image->getSize();
+
+        if ($size->getWidth() < $minWidth || $size->getHeight() < $minHeight) {
+            $errorElement
+                ->with('binaryContent')
+                    ->addViolation('error.image_too_small', [
+                        '%min%' => $minWidth,
+                        '%max%' => $minHeight,
```

Comment:
I guess the validation against max sizes is still in progress.
```suggestion
                        '%min_width%' => $minWidth,
                        '%min_height%' => $minHeight,
```